### PR TITLE
Create approved requests CSV

### PR DESCRIPTION
**Pull request by maxkadel**
Closes: #821

### CODE REVIEW

**Comment by maxkadel** [comment]
CSV generated on staging:

```csv
start_date,end_date,event_name,trip_id,surname,given_name,department,estimated_cost
2020-06-07,2020-06-13,"Open Repositories (OR) 2020, South Africa",10046,Cole,Carolyn Ann,Library Information Technology,2755.00
2020-05-29,2020-06-07,"Arraynavigating 2020, Bryonberg",10077,Powlowski,Jasmin,Library Information Technology,7853.00
2020-08-06,2020-08-15,"Sensorconnecting 2021, Manualside",10079,Powlowski,Jasmin,Library Information Technology,13493.00
2020-06-05,2020-06-08,"Systemindexing 2021, Stromanburgh",10081,Powlowski,Jasmin,Library Information Technology,954.00
2020-08-10,2020-08-20,"Portconnecting 2020, Herbertville",10082,Powlowski,Jasmin,Library Information Technology,6853.00
2021-06-05,2021-06-13,"Programquantifying 2020, Moenstad",10084,Powlowski,Jasmin,Library Information Technology,2290.00
2020-12-09,2020-12-14,"Systemcompressing 2021, Boscostad",10098,Pfeffer,Nedra,Library Information Technology,3142.00
2021-02-19,2021-02-28,"Cardparsing 2021, Rolfsonberg",10141,Watsica,Kittie,Library Information Technology,6183.00
2020-05-23,2020-06-01,"Programcopying 2020, Lake Lamar",10147,Watsica,Kittie,Library Information Technology,9353.00
2020-08-24,2020-09-02,"Panelhacking 2020, Bethannstad",10178,Christiansen,Albertine,Library Information Technology,11089.00
2021-06-12,2021-06-13,"Pixeloverriding 2020, West Aldo",10183,Christiansen,Albertine,Library Information Technology,9708.00
2021-03-07,2021-03-14,"Busquantifying 2021, Virginatown",10213,Pollich,Florentino,Library Information Technology,8486.00
2020-09-28,2020-09-30,"Arraynavigating 2020, South Levibury",10215,Pollich,Florentino,Library Information Technology,4598.00
2020-07-03,2020-07-04,"Microchiphacking 2021, Alannamouth",10216,Pollich,Florentino,Library Information Technology,2868.00
2020-05-29,2020-06-03,"Busoverriding 2020, Port Chasehaven",10268,Labadie,Dan,Library Information Technology,4365.00
2020-10-04,2020-10-08,"Bandwidthquantifying 2020, Birgitview",10278,Labadie,Dan,Library Information Technology,2340.00
2021-04-12,2021-04-16,"Hard Driveparsing 2021, Mosciskiburgh",10279,Labadie,Dan,Library Information Technology,6969.00
2021-03-09,2021-03-18,"Protocolcompressing 2021, Reingerchester",10310,Cartwright,Wilfredo,Library Information Technology,4293.00
2020-11-20,2020-11-25,"Systemnavigating 2020, O'Reillymouth",10330,Herman,Larry,Library Information Technology,8083.00
2021-02-24,2021-02-25,"Firewallparsing 2021, Janfort",10356,Lehner,Carlo,Library Information Technology,6071.00
2021-05-29,2021-05-31,"Protocolcompressing 2020, South Tamramouth",10362,Lehner,Carlo,Library Information Technology,4484.00
2022-07-18,2022-07-21,"Campus Trip 2022, Princeton, NJ",10407,Reiss,Kevin Michael,Library Information Technology,600.00
```

File: app/services/approved_request_report.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @end_date = Date.strptime(end_date, "%m/%d/%Y")
+    @file_path = file_path
+  end
+
+  def csv
+    headers = ["start_date", "end_date", "event_name", "trip_id", "surname", "given_name", "department", "estimated_cost"]
+
+    CSV.open(file_path, "wb") do |csv|
+      csv << headers
+      TravelRequest.find_each do |request|
```
Comment (sandbergja) [inline app/services/approved_request_report.rb:20]: As currently written, this loop runs 3 queries for every row in the CSV file.  Could we eager-load some of the joined values to cut down on the number of SQL queries?

```suggestion
      TravelRequest.includes(:estimates, creator: [:department]).find_each do |request|
```

**Review by sandbergja — CHANGES_REQUESTED** [review]
Thanks @maxkadel !  Just one small suggestion to avoid an N+1 queries situation.

**Comment by maxkadel** [comment]
> Thanks @maxkadel ! Just one small suggestion to avoid an N+1 queries situation.

Thanks @sandbergja! I did write a test to verify this (the log wasn't reflecting the difference) and used https://github.com/civiccc/db-query-matchers to do so - it's pretty neat. I don't know whether it makes sense to add permanently to the test suite, but it was definitely helpful when the log wasn't showing the difference.

**Review by sandbergja — APPROVED** [review]
Thanks, @maxkadel !  And the db-query-matchers gem looks really cool.  Thanks for mentioning that!

**Pull request merged**